src/error-table.rkt

```diff
@@ -134,87 +229,121 @@
                  (bfinfinite? y)
                  (same-sign? x y)
                  (not (bfnan? subexpr-val)))
-            (mark-erroneous! subexpr)]
+            (mark-erroneous! subexpr 'nan-rescue)]
 
            ; inf rescue
            ; If x or y overflow and the other arg rescues
            ; it
-           [(and (or (bfinfinite? x)
-                     (bfinfinite? y))
+           [(and (bfinfinite? x)
+                 (not (bfinfinite? subexpr-val)))
+            (mark-erroneous! subexpr 'oflow-left)]
+           [(and (bfinfinite? y)
                  (not (bfinfinite? subexpr-val)))
-            (mark-erroneous! subexpr)]
+            (mark-erroneous! subexpr 'oflow-right)]
 
            ; High condition number:
            ; CN(+, x, y) = |x / x - y|
            [(or (bf> cond-x cond-thres)
                 (bf> cond-y cond-thres))
-            (mark-erroneous! subexpr)]
+            (mark-erroneous! subexpr 'cancellation)]
+
+           ; Maybe
+           [(or (bf> cond-x maybe-cond-thres)
+                (bf> cond-y maybe-cond-thres))
+            (mark-maybe! subexpr 'cancellation)] 
            [else #f])]
 
-        [(list (or 'sin.f64 'sin.f32) x-ex)
+       [(list (or 'sin.f64 'sin.f32) x-ex)
          #:when (list? x-ex)
          (define x (exacts-ref x-ex))
          (define cot-x (bfabs (bfcot x)))
          (define cond-no (bf* (bfabs x) cot-x))
-
          (cond
+           [(and (bfinfinite? x)
+                 (not (bfnan? subexpr-val)))
+            (mark-erroneous! subexpr 'oflow-rescue)]
            [(and (bf> cond-no cond-thres)
                  (bf> (bfabs x) cond-thres))
-            (mark-erroneous! subexpr)]
+            (mark-erroneous! subexpr 'sensitivity)]
            [(and (bf> cond-no cond-thres)
                  (bf> cot-x cond-thres))
-            (mark-erroneous! subexpr)]
+            (mark-erroneous! subexpr 'cancellation)]
+
+           [(and (bf> cond-no maybe-cond-thres)
+                 (bf> (bfabs x) maybe-cond-thres))
+            (mark-maybe! subexpr 'sensitivity)]
+           [(and (bf> cond-no maybe-cond-thres)
+                 (bf> cot-x maybe-cond-thres))
+            (mark-maybe! subexpr 'cancellation)]
            [else #f])]
 
         [(list (or 'cos.f64 'cos.f32) x-ex)
          #:when (list? x-ex)
```

Comment:
I have been thinking about this. I think it should be a composition of (a) expression that's a product or quotient of constants and (b) representable in floating point. (a) is easy enough not sure how to do (b).

Reply:
I think we should write a little wrapper around `eval-application` and it can check something like:
```
(define (exact-constant? expr)
  (let ([val (eval-application-recursively expr)])
    (and val (= val (exact->inexact val)))))
```

Would be even better to use `real->representation` instead of just `exact->inexact` but that is more complex.